lib/manager.js

```diff
@@ -401,6 +401,7 @@ Manager.prototype.destroy = function(socket){
 Manager.prototype.packet = function(packet){
   debug('writing packet %j', packet);
   var self = this;
+  if (packet.query && packet.type == 0) packet.nsp += '?' + packet.query;
```

Comment:
I wonder this packet can be `{ nsp: '/nsp', data:query }`. so that we don't need to construct and parse query string.